File: topic_5/Haskell/11-rotEsq.hs
-- Recebe uma strig e retorna essa lista rotacionada n vezes à esquerda

rotEsq 0 x = x
rotEsq n x = rotEsq (n-1) esq
    where
        esq = tail x ++ [head x]

main = do
    a <- readLn :: IO Int
    b <- getLine
    print $ rotEsq a b

File: topic_5/Python/11-rotEsq.py
# Recebe uma strig e retorna essa lista rotacionada n vezes à esquerda

def rotEsq(n, x):
    if n == 0:
        return x
    comeco = x[0]
    resto = x[1:]
    esq = resto + comeco
    return rotEsq((n - 1), esq)

print(rotEsq(0, "asdfg") == "asdfg") #True
print(rotEsq(1, "asdfg") == "sdfga") #True
print(rotEsq(3, "asdfg") == "fgasd") #True
print(rotEsq(4, "asdfg") == "gasdf") #True